crates/miden-lib/asm/kernels/transaction/lib/account_delta.masm

```diff
@@ -0,0 +1,186 @@
+use.kernel::memory
+use.kernel::constants
+use.kernel::account
+use.std::crypto::hashes::rpo
+
+# CONSTANTS
+# =================================================================================================
+
+# The domain of an asset in the delta commitment.
+const.DOMAIN_ASSET=1
+# The domain of a value storage slot in the delta commitment.
+const.DOMAIN_VALUE=2
+# The domain of a map storage slot in the delta commitment.
+const.DOMAIN_MAP=3
+
+# PROCEDURES
+# =================================================================================================
+
+#! Computes the commitment to the native account's delta.
+#!
+#! See the Rust function `AccountDelta::commitment` for a detailed description
+#! of how it is computed.
+#!
+#! Inputs:  []
+#! Outputs: [DELTA_COMMITMENT]
+#!
+#! Where:
+#! - DELTA_COMMITMENT is the commitment to the account delta.
+export.compute_commitment
+    # pad capacity element of the hasher
+    padw
+    # => [CAPACITY]
+
+    exec.memory::get_account_nonce_delta push.0
+    # => [0, nonce_delta, CAPACITY]
+
+    exec.memory::get_native_account_id
+    # => [native_acct_id_prefix, native_acct_id_suffix, 0, nonce_delta, CAPACITY]
+    # => [ID_AND_NONCE, CAPACITY]
+
+    padw
+    # => [EMPTY_WORD, ID_AND_NONCE, CAPACITY]
+
+    hperm
+    # => [RATE, RATE, PERM]
+
+    # TODO: Compute vault commitment
+
+    exec.update_storage_delta
+    # => [RATE, RATE, PERM]
+
+    exec.rpo::squeeze_digest
+    # => [DELTA_COMMITMENT]
+end
+
+#! Updates the given delta hasher with the storage slots.
+#!
+#! Inputs:  [RATE, RATE, PERM]
+#! Outputs: [RATE, RATE, PERM]
+proc.update_storage_delta
+    exec.memory::get_num_storage_slots movdn.12
+    # => [RATE, RATE, PERM, num_storage_slots]
+
+    push.0 movdn.12
+    # => [RATE, RATE, PERM, slot_idx = 0, num_storage_slots]
+
+    # loop if num_storage_slots != 0
+    dup.13 neq.0
+    # => [should_loop, RATE, RATE, PERM, slot_idx, num_storage_slots]
+
+    while.true
+        dup.12
+        # => [slot_idx, RATE, RATE, PERM, slot_idx, num_storage_slots]
+
+        exec.update_slot_delta
+        # => [RATE, RATE, PERM, slot_idx, num_storage_slots]
+
+        # increment slot index
+        movup.12 add.1
+        # => [next_slot_idx, RATE, RATE, PERM, num_storage_slots]
+
+        dup movdn.13
+        # => [next_slot_idx, RATE, RATE, PERM, next_slot_idx, num_storage_slots]
+
+        # compute next_slot_idx < num_storage_slots
+        dup.14 lt
+        # => [should_loop, RATE, RATE, PERM, next_slot_idx, num_storage_slots]
+    end
```

Comment:
Maybe not for this PR, but there may be a more efficient way to do it: we could arrange the stack as follows:
```
[RATE, RATE, PERM, slot_ptr, end_ptr]
```
And we'd break the loop on `slot_ptr == end_ptr`. This would give us a couple of benefits:

- Equality comparison is much faster than `lt`.
- We could use `mem_stream` to load the next slot's data onto the stack. This would also give us slot type info basically for free (because we'd load both value and slot metadata).

Two potential complexities with this would be to get slot index and initial slot value based on the `slot_ptr` - but I think these should be pretty easy to handle. We could have procedures like `slot_ptr_to_slot_idx` and `slot_ptr_to_inti_slot_ptr`. Each of these shouldn't require more than a few cycles.